packages/server/api/src/app/flows/flow.module.ts

```diff
@@ -76,6 +87,13 @@ export const flowModule: FastifyPluginAsyncTypebox = async (app) => {
         };
         socket.emit(WebsocketClientEvent.TEST_STEP_FINISHED, response);
       } catch (err) {
+        sendStepFailureEvent({
+          userId: principal?.id ?? '',
+          projectId: principal?.projectId ?? '',
+          stepName: data.stepName,
+          errorMessage: (err as Error).message,
```

Comment:
what is in the errorMessage? what does it look like? I'm concerned this can have sensitive data.